packages/babel-plugin-transform-parameters/src/params.js

```diff
@@ -64,6 +64,31 @@ export default function convertFunctionParams(path, loose) {
   for (let i = 0; i < params.length; i++) {
     const param = params[i];
 
+    for (const name of Object.keys(param.getBindingIdentifiers())) {
+      const constantViolations = scope.bindings[name]?.constantViolations;
```

Comment:
nit: `scope.bindings[name]` is always truthy because param bindings should always be registered in the function scope. And we don't have to empty check `.constantViolations` because it will always be initialized as an empty array. So it suffices to
```js
for (const redeclarator of scope.bindings[name].constantViolations) {}
```

Reply:
```
Summary of all failing tests
 FAIL  packages/babel-plugin-transform-typescript/test/index.js
  ● babel-plugin-transform-typescript/class › parameter properties with parameters

    TypeError: babel/packages/babel-plugin-transform-typescript/test/fixtures/class/parameter-properties-with-parameters/input.ts: Cannot read property 'constantViolations' of undefined
```

Thanks! I tried this out and the above test failed, seems like `scope.bindings[name]` is undefined for the particular case.

Reply:
Don't think I should fix the Typescript part here, so ill leave the checks in for now?